ql/currency.cpp

```diff
@@ -51,7 +52,8 @@ namespace QuantLib {
                        Integer fractionsPerUnit,
                        const Rounding& rounding,
                        const std::string& formatString,
-                       const Currency& triangulationCurrency)
+                       const Currency& triangulationCurrency,
+                       const std::set<std::string>& minorUnitCodes)
```

Comment:
I know that most of the other args are passed by const ref, but how about passing by value and moving into `data_`?

Reply:
Yes, but either for all parameters or none then? Maybe that's worth a different PR?

Reply:
If you don't mind the cost of copying the set in the meantime, then I suppose it can be improved in a separate PR.

Reply:
Yes - that's fine for my purposes anyway. I'd say let's refactor this in a more general PR.

Reply:
clang-tidy is going to open a PR automatically if it detects that passing by value works.